src/main/scala/com/amazon/deequ/analyzers/PatternMatch.scala

```diff
@@ -82,6 +85,19 @@ case class PatternMatch(column: String, pattern: Regex, where: Option[String] =
     conditionalSelection(expression, where).cast(IntegerType)
   }
 
+  private[deequ] def rowLevelResults: Column = {
+    val filteredRowOutcome = getRowLevelFilterTreatment(analyzerOptions)
+    val whereNotCondition = where.map { expression => not(expr(expression)) }
+    val expression = when(regexp_extract(col(column), pattern.toString(), 0) =!= lit(""), 1)
```

Comment:
Can we put `when(regexp_extract(col(column), pattern.toString(), 0) =!= lit(""), 1)` in a common place and reuse it in `criterion` and here?